weblab/entities/forms.py

```diff
@@ -97,6 +97,21 @@ class EntityTagVersionForm(forms.Form):
         required=True)
 
 
+class EntityRenameForm(forms.Form):
+    name = forms.CharField(
+        label='New Name',
+        help_text='',
+        required=True)
+
+    def clean_name(self):
+        name = self.cleaned_data['name']
+        if Entity.objects.filter(name=name).exists():
+            raise ValidationError(
+                'You already have a model or protocol named "%s"' % name)
+
```

Comment:
Ideally would say if it was a model or protocol but unsure how to get the display type here

Reply:
Indeed, it's rather awkward! The trick is used by `EntityCreateView` (see also below) and the two subclasses of `EntityForm` - you basically need a form subclass for each entity type.